src/tbb/thread_data.h

```diff
@@ -251,6 +255,18 @@ inline void thread_data::propagate_task_group_state(std::atomic<std::uint32_t> d
     my_context_list->epoch.store(the_context_state_propagation_epoch.load(std::memory_order_relaxed), std::memory_order_release);
 }
 
+inline void thread_data::set_top_group_task(d1::task* t) {
+    my_task_dispatcher->my_task_group_tasks.push(t);
+}
+
+inline d1::task* thread_data::get_top_group_task() {
+    return my_task_dispatcher->my_task_group_tasks.empty() ? nullptr : my_task_dispatcher->my_task_group_tasks.top();
+}
+
+inline void thread_data::remove_top_group_task() {
+    my_task_dispatcher->my_task_group_tasks.pop();
```

Comment:
Consider adding an assertion that `!my_task_group_tasks.empty()` since `std::deque::pop_back()` results in UB for empty container.